spring-integration-amqp/src/main/java/org/springframework/integration/amqp/support/AmqpDedicatedChannelAdvice.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.amqp.support;
+
+import java.time.Duration;
+
+import org.aopalliance.intercept.MethodInvocation;
+
+import org.springframework.amqp.rabbit.core.RabbitOperations;
+import org.springframework.integration.handler.advice.HandleMessageAdvice;
+import org.springframework.lang.Nullable;
+
+/**
+ * An advice that executes all downstream {@link RabbitOperations} operations
+ * on the same channel, as long as there are no thread handoffs.
+ * The same RabbitOperations must be used in this and all downstream components.
+ * Typically used with a splitter or some other mechanism that would cause
+ * multiple messages to be sent.
+ *
+ * @author Gary Russell
+ * @since 5.1
+ *
+ */
+public class AmqpDedicatedChannelAdvice implements HandleMessageAdvice {
+
+	private final RabbitOperations operations;
+
+	private final Duration waitForConfirmsTimeout;
+
+	/**
+	 * Construct an instance that doesn't wait for confirms.
+	 * @param operations the operations.
+	 */
+	public AmqpDedicatedChannelAdvice(RabbitOperations operations) {
+		this(operations, null);
+	}
+
+	/**
+	 * Construct an instance that waits for publisher confirms (if
+	 * configured and waitForConfirmsTimeout is not null).
+	 * @param operations the operations.
+	 * @param waitForConfirmsTimeout the timeout.
+	 */
+	public AmqpDedicatedChannelAdvice(RabbitOperations operations, @Nullable Duration waitForConfirmsTimeout) {
+		this.operations = operations;
+		this.waitForConfirmsTimeout = waitForConfirmsTimeout;
+	}
+
+	@Override
+	public Object invoke(MethodInvocation invocation) throws Throwable {
+		try {
+			return this.operations.invoke(operations -> {
+				try {
+					Object result = invocation.proceed();
+					if (this.waitForConfirmsTimeout != null) {
+						this.operations.waitForConfirmsOrDie(this.waitForConfirmsTimeout.toMillis());
+					}
+					return result;
+				}
+				catch (Throwable t) { // NOSONAR - rethrown below
+					throw new ThrowableWrapper(t);
```

Comment:
`ReflectionUtils.rethrowRuntimeException()` ?

Reply:
? 

`proceed()` throws `Throwable`.

Reply:
?
```
public static void rethrowRuntimeException(Throwable ex) {
```
Then in the main `catch` you can use `UndeclaredThrowableException.getUndeclaredThrowable()`

Reply:
Oh - I was looking at the wrong `rethrow...` method.

Fixing.